Docs--add-content-block-page-v2 (#1750)
carbon-design-system/carbon-for-ibm-dotcom-website: src/pages/components/content-block.mdx

---

import ComponentDescription from 'components/ComponentDescription';
import ComponentFooter from 'components/ComponentFooter';
import ResourceLinks from 'components/ResourceLinks';

<ComponentDescription name="Content block" type="layout" />

<AnchorLinks>

<AnchorLink>Overview</AnchorLink>
<AnchorLink>Modifiers</AnchorLink>
<AnchorLink>Behaviors</AnchorLink>
<AnchorLink>Gallery</AnchorLink>
<AnchorLink>Resources</AnchorLink>
<AnchorLink>Content guidance</AnchorLink>
<AnchorLink>Related components</AnchorLink>
<AnchorLink>Feedback</AnchorLink>

</AnchorLinks>

## Overview

The content block acts as the heading level two on pages after the lead space component and spans 12 columns of the 16 column grid. The content block is commonly used to divide the main categories or sections of the page to help guide the user through the content or narrative.

![Image of the content block with a heading, subheading, body copy, link with item, and bottom border](../../images/component/content-block/content-block-overview.png)

<Caption>Example of the content block.</Caption>

### Anatomy

![Anatomy image of the content block](../../images/component/content-block/content-block-anatomy.png)

> 1. **Content block heading:** An optional, customizable heading for the Content section.

Reviewer: ```suggestion
1. **Content block heading:** An optional, customizable heading for the content block.
```

> 2. **Sub heading:** An optional, customizable sub heading for the Content section.

Reviewer: ```suggestion
2. **Sub heading:** An optional, customizable sub heading for the content block.
```

3. **Copy:** An optional short description.
4. **Child container:** Insert additional components to enhance the narrative, such as content group, content item or card group.

> 5. **CTA:** Use this as a general call to action, typically used to encompass the entire content section.

Reviewer: ```suggestion
5. **CTA:** Use this as a general call to action, typically used to encompass the entire content block.
```

![A content block with a heading and body copy spanning 8 columns. Nested inside the content block is three content items spanning 12 columns.](../../images/component/content-block/content-block-modifiers-03.png)

<Caption>
  Example of the content block with a heading and subheading spanning 8 columns,
  and a set of content items spanning 12 columns.
</Caption>

## Behaviors

At smaller breakpoints, if content block is paired with a content section, the content section heading stacks on top of the content block heading.

<Row>

<Column colMd={8} colLg={8}>

![Image of the content block at tablet breakpoint](../../images/component/content-section/content-section-behavior-01.png)

<Caption>
  Example of content block with a content section heading at medium breakpoint
</Caption>

</Column>

<Column colMd={8} colLg={4}>

![Image of the content section at mobile breakpoint](../../images/component/content-section/content-section-behavior-02.png)

<Caption>
  Example of content block with a content section heading at mobile breakpoint
</Caption>

</Column>

</Row>

## Gallery

> The content block is one of the main layout components for designing pages of the IBM.com platform. Here are some examples of various layouts you can create by utilizing the extreme flexibility the content section provides by utilizing the varying heading options and adding custom children.

Reviewer: ```suggestion
The content block is one of the main layout components for designing pages of the IBM.com platform. Here are some examples of various layouts you can create by utilizing the extreme flexibility the content block provides by utilizing the varying heading options and adding custom children.
```

---

import ComponentDescription from 'components/ComponentDescription';
import ComponentFooter from 'components/ComponentFooter';
import ResourceLinks from 'components/ResourceLinks';

<ComponentDescription name="Content block" type="layout" />

<AnchorLinks>

<AnchorLink>Overview</AnchorLink>
<AnchorLink>Modifiers</AnchorLink>
<AnchorLink>Behaviors</AnchorLink>
<AnchorLink>Gallery</AnchorLink>
<AnchorLink>Resources</AnchorLink>
<AnchorLink>Content guidance</AnchorLink>
<AnchorLink>Related components</AnchorLink>
<AnchorLink>Feedback</AnchorLink>

</AnchorLinks>

## Overview

The content block acts as the heading level two on pages after the lead space component and spans 12 columns of the 16 column grid. The content block is commonly used to divide the main categories or sections of the page to help guide the user through the content or narrative.

![Image of the content block with a heading, subheading, body copy, link with item, and bottom border](../../images/component/content-block/content-block-overview.png)

> <Caption>Example of the content section.</Caption>

Reviewer: ```suggestion
<Caption>Example of the content block.</Caption>
```

### Anatomy

![Anatomy image of the content block](../../images/component/content-block/content-block-anatomy.png)

1. **Content block heading:** An optional, customizable heading for the Content section.
2. **Sub heading:** An optional, customizable sub heading for the Content section.
3. **Copy:** An optional short description.
4. **Child container:** Insert additional components to enhance the narrative, such as content group, content item or card group.
5. **CTA:** Use this as a general call to action, typically used to encompass the entire content section.
6. **Border:** An optional bottom border.

### Usage

> Content block is often paired with the content section or the vertical table of contents to guide the user through the narrative and activate the first four columns of the 16 column grid.

Reviewer: ```suggestion
Content block is often paired with the [content section](../components/content-section) or the [vertical table of contents](../components/table-of-contents#vertical) to guide the user through the narrative and activate the first four columns of the 16 column grid.
```

<Row>

<Column colMd={8} colLg={6}>

![Content block with a section heading at the desktop breakpoint. The section heading is highlighted.](../../images/component/content-block/content-block-usage-01.png)

<Caption>
  Content block with a header and content group containing content items
  featuring pictograms
</Caption>

</Column>

<Column colMd={8} colLg={6}>

![Content block with a table of contents - vertical  at the desktop breakpoint. The table of contents - vertical is highlighted.](../../images/component/content-block/content-block-usage-02.png)

<Caption>
  Content block with a header and card group child spanning 12 columns
</Caption>

</Column>

</Row>

## Modifiers

> Content block is often paired with the content section or the vertical table of contents to guide the user through the narrative and activate the first four columns of the 16 column grid.

Reviewer: ```suggestion
```

Remove this section, duplicate of "Usage" above

### Children

Similar to the content section, the content block can accept child components which allows designers and authors maximum flexibility when creating page layouts. The most common layouts on IBM.com are 8 column and 12 column components.

> There is no limit to how many child components the content section can accept - consider the story you are trying to tell, the overall hierarchy of the page and how many content sections should be used to break up the content. View the [gallery](#gallery) to see examples of content section with various child components.

Reviewer: ```suggestion
There is no limit to how many child components the content block can accept - consider the story you are trying to tell, the overall hierarchy of the page and how many content blocks should be used to break up the content. View the [gallery](#gallery) to see examples of content block with various child components.
```

#### 8 column child

> In this example the content section is paired with a group of [content item pictograms](./content-item) that span the center 8 columns of the grid.

Reviewer: ```suggestion
In this example the content block is paired with a group of [content item pictograms](./content-item) that span the center 8 columns of the grid.
```

![A content block with only the heading enabled. Nested inside the content block is three content items in the horizontal orientation, each displaying a different pictogram and contianing different content.](../../images/component/content-block/content-block-modifiers-01.png)

<Caption>
  Content block with a heading and content group containing content items
  featuring pictograms
</Caption>

#### 12 column child

> In this example the content section is paired with a [card group](./card-group) that spans 12 columns of the grid.

Reviewer: ```suggestion
In this example the content block is paired with a [card group](./card-group) that spans 12 columns of the grid.
```
